src/DynamoRevit/Models/RevitDynamoModel.cs

```diff
@@ -611,7 +617,7 @@ private void OnResetMarkNodesAsDirty(bool markNodesAsDirty)
                 workspace.ResetEngine(EngineController, markNodesAsDirty);
         }
 
-        public void SetRunEnabledBasedOnContext(View newView)
+        public void SetRunEnabledBasedOnContext(View newView, bool doContextAvailable)
```

Comment:
since this is a public method (and lets say this is in the API... still up for debate) we should either add a new method and obsolete the old one, or simpler make doContextAvailable an optional parameter.